File: frontend/src/types/song.ts
export interface Track {
  id: string;
  name: string;
  section: string;
  icon: string;
  audioFile: string;
  scoreFile?: string;
}

export interface Song {
  slug: string;
  title: string;
  composer: string;
  description: string;
  category: 'classical' | 'folk';
  icon: string;
  iconColor: string;
  totalAudio: string;
  totalScore?: string;
  tracks: Track[];
}

File: backend/app/schemas/song.py
"""
Song schemas.
"""
from pydantic import BaseModel


class Track(BaseModel):
    """Track schema."""

    id: str
    name: str
    section: str
    icon: str
    audioFile: str
    scoreFile: str | None = None


class Song(BaseModel):
    """Song schema."""

    slug: str
    title: str
    composer: str
    description: str
    category: str
    icon: str
    iconColor: str
    totalAudio: str
    totalScore: str | None = None
    tracks: list[Track]


class SongListResponse(BaseModel):
    """Song list response."""

    songs: list[Song]
    total: int


class SongDetailResponse(BaseModel):
    """Song detail response."""

    song: Song
